lib/queuePopulator/QueuePopulator.js

```diff
@@ -109,6 +114,19 @@ class QueuePopulator {
         this._mProducer.setupProducer(cb);
     }
 
+    /**
+     * Set up the consumer and producer for retrying failed CRR operations.
+     * @param {Function} cb - The callback function
+     * @return {undefined}
+     */
+    _setupRetryClients(cb) {
+        this._retryConsumer = new RetryConsumer(this.kafkaConfig);
```

Comment:
Let's try not to stretch the role of queue populator beyond its metadata ingestion scope.

The producer is fine in the replication status processor IMO because that's where the info is readily available to be published.

The consumer should be part of its own component updating redis for separation of concerns.

Reply:
I'm not sure I follow what separating the start of the consumer it into its own component would look like. Should it be a backbeat "extension" with its own task? Maybe a good place is https://github.com/scality/backbeat/blob/master/extensions/replication/queueProcessor/task.js#L49. In any case, I noticed the metrics consumer was being started when opening the queue populator so added it here.

Reply:
The idea is to follow a philosophy where each backbeat internal component should be runnable as an independent entity. All entities that belong to the same service (e.g. replication) will go under a single kub service. The general idea as I understand it is to make it more manageable, pluggable on-demand and scalable independently from each other.

In this case, I think it makes sense to create a new component for consuming the retry queue and updating redis. In that case, indeed there should be an entrypoint that starts the process (task.js or service.js), and a set of classes doing the job (e.g. in tasks/ for the processing triggered by consumption of a single kafka entry).

I think it was an overlook (or a shortcut) to put the metrics consumer in the queue populator, it should only contain the producer, and the consumer should be a separate process.